server/lib/realtime/interpreter/transient.ex

```diff
@@ -0,0 +1,172 @@
+  end
+
+  @impl true
+  def handle_continue({:start, ctx, args}, %State{workflow: workflow} = state) do
+    Workflows.start(workflow, ctx, args)
+    |> continue_with_result(state)
+  end
+
+  def handle_continue(:continue_process_events, state) do
+    process_next_event()
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_cast(
+        :process_next_event,
+        %State{execution_id: execution_id, events: events} = state
+      ) do
+    case events do
+      [] ->
+        # Finished processing all events, wait for side effects to complete
+        {:noreply, state}
+
+      [event | events] ->
+        :ok = execute_event_side_effect(event, execution_id)
+        {:noreply, %State{state | events: events}, {:continue, :continue_process_events}}
+    end
+  end
+
+  @impl true
+  def handle_info({:finish_waiting, event}, %State{execution: execution} = state) do
+    command = Command.finish_waiting(event)
+
+    Workflows.resume(execution, command)
+    |> continue_with_result(state)
+  end
+
+  def handle_info({:complete_task, event, result}, %State{execution: execution} = state) do
+    command = Command.complete_task(event, result)
+
+    Workflows.resume(execution, command)
+    |> continue_with_result(state)
+  end
+
+  ## Private
+
+  defp process_next_event() do
+    GenServer.cast(self(), :process_next_event)
+  end
+
+  defp execute_event_side_effect(%Event.WaitStarted{} = event, execution_id) do
+    duration =
+      case event.wait do
+        {:seconds, seconds} when is_integer(seconds) and seconds > 0 ->
+          trunc(seconds * 1000)
+
+        {:timestamp, target} ->
+          DateTime.diff(target, DateTime.utc_now(), :millisecond)
+
+        _ ->
+          {:error, "Invalid wait duration"}
+      end
+
+    Process.send_after(self(), {:finish_waiting, event}, duration)
+    :ok
+  end
+
+  defp execute_event_side_effect(%Event.TaskStarted{} = event, execution_id) do
+    server_pid = self()
+
+    Task.start(fn ->
+      # TODO: what to do in case of error?
+      res = handle_event(event, execution_id)
+      send(server_pid, {:complete_task, event, %{}})
+    end)
+
+    :ok
+  end
+
+  defp execute_event_side_effect(_event, _execution_id) do
+    :ok
+  end
+
+  defp continue_with_result({:continue, execution, events}, state) do
+    new_state = %State{
+      state
+      | execution: execution,
+        events: state.events ++ events
+    }
+
+    Logger.info("continue HERE")
+
+    {:noreply, new_state, {:continue, :continue_process_events}}
+  end
+
+  defp continue_with_result({:succeed, result, events}, state) do
+    Logger.info("Execution terminated with result #{inspect(result)} #{inspect(self())}")
+
+    if is_pid(state.reply_to) do
+      send(state.reply_to, {:succeed, result})
+    end
+
+    {:stop, :normal, state}
+  end
+
+  defp handle_event(%Event.TaskStarted{} = event, execution_id) do
+    # TODO: Do we need Oban backing for transient workflows? Might be good for tracking failures
```

Comment:
@fracek - need to check with you on the implementation you were thinking for transient workflows. Does it require Oban?

Reply:
I think it's fine (maybe event better) without Oban. I was thinking to use Oban to gain observability of workflows executions, but I think logging the errors to AWS is a better choice in general.

Reply:
awesome thanks. I'm thinking for this early version we should try to touch the database as little as possible. once we know it's scaling we can add in a bit more observability.